src/physalis.h

```diff
@@ -56,75 +56,147 @@
-#define SOUTH_WALL -12
+#define EAST_WALL_D -110
 /* FUNCTION
- *  Define a value to use for identifying that the SOUTH wall is interfering with
- *  a quadrature node
+ *  Define a value to use for identifying that the east wall(dirichlet) is
+ *  interfering with a quadrature node
  ******
  */
 
-/****d* physalis/NORTH_WALL
+/****d* physalis/EAST_WALL_N
  * NAME
- *  NORTH_WALL
+ *  EAST_WALL_N
  * USAGE
  */
-#define NORTH_WALL -13
+#define EAST_WALL_N -111
 /* FUNCTION
- *  Define a value to use for identifying that the NORTH wall is interfering with
- *  a quadrature node
+ *  Define a value to use for identifying that the east wall(neumann) is
+ *  interfering with a quadrature node
  ******
  */
 
-/****d* physalis/BOTTOM_WALL
+/****d* physalis/SOUTH_WALL_D
  * NAME
- *  BOTTOM_WALL
+ *  SOUTH_WALL_D
  * USAGE
  */
-#define BOTTOM_WALL -14
+#define SOUTH_WALL_D -120
 /* FUNCTION
- *  Define a value to use for identifying that the BOTTOM wall is interfering with
- *  a quadrature node
+ *  Define a value to use for identifying that the SOUTH wall(dirichlet)
+ *  is interfering with a quadrature node
  ******
  */
 
-/****d* physalis/TOP_WALL
+/****d* physalis/SOUTH_WALL_N
  * NAME
- *  TOP_WALL
+ *  SOUTH_WALL_N
  * USAGE
  */
-#define TOP_WALL -15
+#define SOUTH_WALL_N -121
 /* FUNCTION
- *  Define a value to use for identifying that the TOP wall is interfering with
- *  a quadrature node
+ *  Define a value to use for identifying that the SOUTH wall(neumann)
+ *  is interfering with a quadrature node
+ ******
+ */
+
+/****d* physalis/NORTH_WALL_D
+ * NAME
+ *  NORTH_WALL_D
+ * USAGE
+ */
+#define NORTH_WALL_D -130
+/* FUNCTION
+ *  Define a value to use for identifying that the NORTH wall(dirichlet)
+ *  is interfering with a quadrature node
+ ******
+ */
+
+/****d* physalis/NORTH_WALL_N
+ * NAME
+ *  NORTH_WALL_N
+ * USAGE
+ */
+#define NORTH_WALL_N -131
+/* FUNCTION
+ *  Define a value to use for identifying that the NORTH wall(neumann)
+ *  is interfering with a quadrature node
+ ******
+ */
+
+/****d* physalis/BOTTOM_WALL_D
+ * NAME
+ *  BOTTOM_WALL_D
+ * USAGE
+ */
+#define BOTTOM_WALL_D -140
+/* FUNCTION
+ *  Define a value to use for identifying that the BOTTOM wall(dirichlet)
+ *  is interfering with a quadrature node
+ ******
+ */
+
+/****d* physalis/BOTTOM_WALL_N
+ * NAME
+ *  BOTTOM_WALL_N
+ * USAGE
+ */
+#define BOTTOM_WALL_N -141
+/* FUNCTION
+ *  Define a value to use for identifying that the BOTTOM wall(neumann)
+ *  is interfering with a quadrature node
+ ******
+ */
+
+/****d* physalis/TOP_WALL_D
+ * NAME
+ *  TOP_WALL_D
+ * USAGE
+ */
+#define TOP_WALL_D -150
+/* FUNCTION
+ *  Define a value to use for identifying that the TOP wall(dirichlet)
+ *  is interfering with a quadrature node
+ ******
+ */
+
+/****d* physalis/TOP_WALL_N
+ * NAME
+ *  TOP_WALL_N
+ * USAGE
+ */
+#define TOP_WALL_N -151
+/* FUNCTION
+ *  Define a value to use for identifying that the TOP wall(neumann)
+ *  is interfering with a quadrature node
```

Comment:
Interesting - so you changed how we deal with quadrature nodes depending on if they intersect with a Neumann vs. Dirichlet-type wall. What do you different? What was the bug / issue that caused you to do this?

Reply:
I implemented temperature field based on the implementation of pressure. I remember that the pressure at lebdev nodes are set to be zero if it is outside domain. But if I did the same thing for temperature, then the temperature field around those near-Dirichlet-boundary particles are almost always very small. And the energy is not conserved. So for Dirichlet boundary, I set the temperature at lebdev nodes to be the boundary temperature and for Neumann boundary, I set it to be the temperature in neighboring cell. That's why I need to differentiate those two.